website/docs/reference/resource-properties/constraints.md

```diff
@@ -518,3 +518,73 @@ alter table schema_name.my_model add constraint 472394792387497234 check (id > 0
 </div>
 
 </WHCode>
+
+### Custom constraints on models for advanced configuration of tables
+
+In dbt Cloud and dbt Core, you can use custom constraints on models for the advanced configuration of tables. Different data warehouses support different syntax and capabilities. However, custom constraints allow you to add configuration to specific columns.
+
+  - For example, custom constraints allow you to set [masking policies](https://docs.snowflake.com/en/user-guide/security-column-intro#what-are-masking-policies) in Snowflake when using a Create Table As Select (CTAS).
+  
+  - [Databricks](https://docs.databricks.com/en/sql/language-manual/sql-ref-syntax-ddl-create-table-using.html) and [BigQuery](https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#column_name_and_column_schema) are other examples of data warehouses having their own set of parameters that can be set for columns in their CTAS statements.
+
+
+You can implement constraints in a couple of different ways:
```

Comment:
it might be helpful to provide users with info on when or why they would implement with tags or without tags?  like what are the benefits of either? maybe @b-per can provide some context on why a user would implement one over the other?

Reply:
Hiya @mirnawong1 @b-per 

Apologies for the delay in getting back to you both on this. I was having a look into the benefits of implementing custom constraints with tags and found they offer users: improvement of data quality, aids organisation, and allows for scalable, reusable, and flexible management of validation rules across projects.

And the benefits of using custom constraints without tags offers simplicity (easier debugging and no overheads in the sense that constraints are applied directly to specific models or columns), greater control, and minimal project complexity are priorities.

I can add this to each expandable section if you're happy for me to do so?

Kind Regards
Natalie

Reply:
I think I mentioned it before, but I don't think that we need to give too much explanation on tags. This is a data warehouse more than a dbt feature. I just want people already interested in applying tags to know how to do it in dbt, not to teach people that they might want to use tags.